db/migrate/20200527161505_create_page_redirect.rb

```diff
@@ -0,0 +1,17 @@
+class CreatePageRedirect < ActiveRecord::Migration[5.2]
+  def change
+    create_table :page_redirects do |t|
+      t.string :old_path, null: false
+      t.string :new_path, null: false
+      t.string :source, null: false, default: "service"
+      t.integer :version, null: false, default: 0
+      t.timestamps null: false
+    end
+
+    add_index :page_redirects, :old_path, unique: true
+    add_index :page_redirects, :new_path
+    add_index :page_redirects, %i[old_path new_path], unique: true
```

Comment:
I don't think we need this. `old_path` is either unique on the entire table, or unique in the context of `new_path`. Don't know which should be, depends on your intentions.

Reply:
My thinking here was for when we run a query such as:
```ruby
Redirect.find_by(old_path: "/some-old-path", new_path: "/some-new_path")
```
Which we aren't doing yet. And maybe that index wouldn't even help with that?

Reply:
Ah got it! Ok, so the third one is definitely redundant then. Because PostgreSQL is smart enough to use the two indexes you already have on line 11 and 12. 

Two options:

- one composite unique index on both old_path and new_path (but this would not make old_path unique per se, just in relation of new path)
- the two indexes without the third

Reply:
Makes sense - I forgot that PostgreSQL knows to use the first 2! Deleted the third index 💯